mlrun/hub/module.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import sys
+from pathlib import Path
+from typing import Optional, Union
+
+import yaml
+
+import mlrun.common.types
+import mlrun.utils
+from mlrun.common.schemas.hub import HubSourceType
+from mlrun.run import function_to_module, get_object
+
+from ..model import ModelObj
+from ..utils import extend_hub_uri_if_needed
+
+
+class ModuleType(mlrun.common.types.StrEnum):
+    generic = "generic"
+    monitoring_app = "monitoring-app"
+
+
+class HubModule(ModelObj):
+    def __init__(
+        self,
+        name: str,
+        kind: Union[ModuleType, str],
+        version: Optional[str] = None,
+        description: Optional[str] = None,
+        categories: Optional[list] = None,
+        requirements: Optional[list] = None,
+        local_path: Optional[str] = None,
+        filename: Optional[str] = None,
+        example: Optional[str] = None,
+        url: Optional[str] = None,
+        **kwargs,  # catch all for unused args
+    ):
+        self.name: str = name
+        self.version: str = version
+        self.kind: ModuleType = kind
+        self.description: str = description or ""
+        self.categories: list = categories or []
+        self.requirements: list = requirements or []
+        self.local_path: str = local_path or ""
+        self.filename: str = filename or name + ".py"
+        self.example: str = example or ""
+        self.url: str = url or ""
+
+    def module(self):
+        try:
+            return function_to_module(self.filename, self.local_path)
```

Comment:
just a recommendation to use keywords argument, it's making it more clear for someone who is not familiar with these apis. 
```suggestion
            return function_to_module(code=self.filename, workdir=self.local_path)
```